lib/Dialect/SPIRV/SPIRVDialect.cpp

```diff
@@ -171,11 +184,38 @@ static Type parseArrayType(SPIRVDialect const &dialect, StringRef spec,
     return Type();
   }
 
+  ArrayType::LayoutInfo layoutInfo = 0;
+  size_t lastLSquare;
+
+  // Handle case when element type is not a trivial type
+  auto lastRDelimiter = spec.rfind('>');
+  if (lastRDelimiter != StringRef::npos) {
+    lastLSquare = spec.find('[', lastRDelimiter);
+  } else {
+    lastLSquare = spec.rfind('[');
+  }
+
+  // The check is relaxed in the parser
```

Comment:
What does this comment mean? Can we change it to something more suitable?

Reply:
Thanks, I've delete it.